src/query/storages/fuse/src/operations/common/processors/transform_mutation_aggregator.rs

```diff
@@ -216,6 +210,10 @@ impl TableMutationAggregator {
     pub async fn apply(&mut self) -> Result<CommitMeta> {
         let appended_segments = std::mem::take(&mut self.appended_segments);
         let appended_statistics = std::mem::take(&mut self.appended_statistics);
+
+        let mut new_segment_locs = Vec::new();
+        new_segment_locs.extend(appended_segments.clone());
```

Comment:
Here, I'm puzzled:

Although it seems like `appended_segments` should be appended to `new_segment_locs`, before this PR, only new segments of `MutationKind` that are not `Insert` were collected into the `abort_operation` (hope I get it correctly).

Am I missing something here? 

should `append_segments` have been collected into the `abort_operation` in the previous versions?
or is it related to the removal of  `self.abort_operation.merge(abort_operation)` while processing mutation log entry of kind `MutationLogEntry::AppendSegment`?

Reply:
> should `append_segments` have been collected into the `abort_operation` in the previous versions? or is it related to the removal of `self.abort_operation.merge(abort_operation)` while processing mutation log entry of kind `MutationLogEntry::AppendSegment`?

It was previously collected in the `abort_operation` of `MutationLogEntry::AppendSegment`, the code is located https://github.com/datafuselabs/databend/blob/v1.2.470-nightly/src/query/storages/fuse/src/operations/common/processors/transform_serialize_segment.rs#L203-L207

And then merged into the `abort_operation` of `TransformSerializeSegment`